### PR TITLE
implement proposed BatteryCharging method for energy estimation charged in a fixed period

### PR DESCRIPTION
**Pull request by nkuehnel**

### CODE REVIEW

File: contribs/ev/src/main/java/org/matsim/contrib/ev/charging/FastThenSlowCharging.java [hunk 1]
```diff
@@ -54,6 +54,47 @@ public double calcChargingPower(double maxPower) {
 		}
 	}
 
+	@Override
+	public double calcEnergyCharge(ChargerSpecification charger, double chargePeriod) {
+		double remainingTime = chargePeriod;
+
+		double maxChargingPower = charger.getPlugPower();
+		double powerCharged = 0;
```
Comment (michalmac) [inline contribs/ev/src/main/java/org/matsim/contrib/ev/charging/FastThenSlowCharging.java:62]: I guess you mean `energyCharged`

File: contribs/ev/src/main/java/org/matsim/contrib/ev/charging/FastThenSlowCharging.java [hunk 2]
```diff
@@ -54,6 +54,47 @@ public double calcChargingPower(double maxPower) {
 		}
 	}
 
+	@Override
+	public double calcEnergyCharge(ChargerSpecification charger, double chargePeriod) {
```
Comment (michalmac) [inline contribs/ev/src/main/java/org/matsim/contrib/ev/charging/FastThenSlowCharging.java:58]: This class does a bit more than a simple mathematical operation, so please add some tests for this methods (there are some tests for the other method, maybe you can base on them).

**Review by michalmac — APPROVED** [review]
Thanks! LGTM

Added a few comments to address.

File: contribs/ev/src/main/java/org/matsim/contrib/ev/charging/FastThenSlowCharging.java (hunk 1 above)
Comment (nkuehnel) [reply contribs/ev/src/main/java/org/matsim/contrib/ev/charging/FastThenSlowCharging.java]: fixed in 4dc3a81

File: contribs/ev/src/main/java/org/matsim/contrib/ev/charging/FastThenSlowCharging.java (hunk 2 above)
Comment (nkuehnel) [reply contribs/ev/src/main/java/org/matsim/contrib/ev/charging/FastThenSlowCharging.java]: You are right! I added a test (and indeed there were som ebugs ;) ) thanks! fixed in 672afbf

**Pull request merged**